### ISSUE TITLE
segmentation fault on OcilibConnectionFree

### ISSUE THREAD

**Issue by jkstill** [vrogier/ocilib#366]
After I started this, I found the issue is due to an oracle bug.

**Bug 33726788 - OCI Unregister Is Stuck During DB Drain Rolling Shutdown (Doc ID 33726788.8)**

This only happens in RAC.  When I switched to a non-RAC 21c database, the issue was not present.

It seemed a good idea to have this issue present for anyone else that searches on the problem.

Please just mark it as resolved, or, delete if  you disagree :)

=======================================

Environment: 
ocilib-4.7.6
Oracle 19.12 on Linux 7.9

$ uname -r
4.14.35-2047.510.5.3.el7uek.x86_64

I compiled a couple of the demo programs, ocilib_demo.c and scroll.c
Both get segmentation fault when OCI_ConnectionFree is called.

I ran scroll with gdb, and set breakpoints on OcilibConnectionFree, CHECK_PTR and LIST_ATOMIC_REMOVE

Results from GDB:
```

59 - Row 59
60 - Row 60
45 - Row 45


Calling OCI_StatementFree(st)

Calling OCI_ConnectionFree(cn)


Breakpoint 1, OcilibConnectionFree (con=0x6bec18) at connection.c:1195
1195    {
Missing separate debuginfos, use: debuginfo-install glibc-2.17-325.0.1.el7_9.x86_64 libaio-0.3.109-13.el7.x86_64 libgcc-4.8.5-44.0.3.el7.x86_64 numactl-libs-2.0.12-5.0.3.el7.x86_64
(gdb) break CHECK_PTR
Function "CHECK_PTR" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 2 (CHECK_PTR) pending.
(gdb) break LIST_ATOMIC_REMOVE
Function "LIST_ATOMIC_REMOVE" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 3 (LIST_ATOMIC_REMOVE) pending.
(gdb) c
Continuing.

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff522efa6 in kpunOps () from /u01/app/oracle/product/19.0.0/dbhome_1/lib/libclntsh.so.19.1
```

**Comment by cjbj** [comment]
That particular bug is marked as fixed in DB 23.1, and the fix was backported to 19.16+. I don't see it marked as backported to 21c (which was an innovation release).

**Comment by vrogier** [comment]
Hi,

This issue is related to OCI client, not OCILIB.
I will then label it accordingly and close it.

@cjbj  thanks for the information :)

Regards,

Vincent
